src/broadcast.jl

```diff
@@ -41,11 +41,11 @@ for f in libdevice
 end
 
 #broadcast ^
-culiteral_pow(::typeof(^), x::Union{Float32, Float64}, ::Val{0}) = one(x)
-culiteral_pow(::typeof(^), x::Union{Float32, Float64}, ::Val{1}) = x
-culiteral_pow(::typeof(^), x::Union{Float32, Float64}, ::Val{2}) = x*x
-culiteral_pow(::typeof(^), x::Union{Float32, Float64}, ::Val{3}) = x*x*x
-culiteral_pow(::typeof(^), x::Union{Float32, Float64}, ::Val{p}) where p = CUDAnative.pow(x, Int32(p))
+culiteral_pow(::typeof(^), x::T, ::Val{0}) where {T<:Real} = one(x)
+culiteral_pow(::typeof(^), x::T, ::Val{1}) where {T<:Real} = x
+culiteral_pow(::typeof(^), x::T, ::Val{2}) where {T<:Real} = x * x
+culiteral_pow(::typeof(^), x::T, ::Val{3}) where {T<:Real} = x * x * x
+culiteral_pow(::typeof(^), x::T, ::Val{p}) where {T<:Real,p} = CUDAnative.pow(x, Int32(p))
```

Comment:
How does this add anything over the original implementation?
[CUDAnative.pow seems to only accept Float32/64 anyway](https://github.com/JuliaGPU/CUDAnative.jl/blob/9298b61a9b9ca5c84939eb6d59d0114bde7a1a27/src/device/cuda/math.jl#L191), how is this any different?

Reply:
This might help to understand.
https://github.com/JuliaDiff/ForwardDiff.jl/blob/958711d983a3f4a18521b846c4cec9219137eb0d/src/dual.jl#L469-L479

Reply:
Won't the last call still break though?
The call to CUDAnative won't be found if `! T<:Union{Float32, Float64}`